File: docs/examples/objc-api-walkthrough/objc-api-walkthrough/ExampleConflictResolver.m
//
//  ExampleConflictResolver.m
//  objc-api-walkthrough
//
//  Created by James Nocentini on 27/07/2017.
//  Copyright © 2017 couchbase. All rights reserved.
//

#import "ExampleConflictResolver.h"

@implementation ExampleConflictResolver

- (nullable CBLDocument*) resolve: (CBLConflict*)conflict {
    CBLDocument* base = conflict.base;
    CBLDocument* mine = conflict.mine;
    CBLDocument* theirs = conflict.theirs;
    return theirs;
}

@end

File: docs/examples/swift-api-walkthrough/api-walkthrough/ExampleConflictResolver.swift
//
//  ExampleConflictResolver.swift
//  api-walkthrough
//
//  Created by James Nocentini on 27/07/2017.
//  Copyright © 2017 couchbase. All rights reserved.
//

import Foundation
import CouchbaseLiteSwift

class ExampleConflictResolver: ConflictResolver {
    func resolve(conflict: Conflict) -> Document? {
        let base = conflict.base
        let mine = conflict.mine
        let theirs = conflict.theirs
        return theirs
    }
}
